### PR TITLE
 [compiler] Add SStackStruct, use to implement MakeStruct

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by tpoterba** [comment]
Benchmarks show that this doesn't change performance, with the exception of making linear_regression_rows_nd about 4x slower. the unreliability of ndarray performance is being solved separately.

**Comment by tpoterba** [comment]
ready for review!

File: hail/src/main/scala/is/hail/types/physical/stypes/concrete/SStackStruct.scala [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+  def store(cb: EmitCodeBuilder, pv: SCode): Unit = {
+    val ssc = pv.asInstanceOf[SStackStructCode]
+    settables.zip(ssc.codes).foreach { case (s, c) => s.store(cb, c) }
+  }
+}
+
+class SStackStructCode(val st: SStackStruct, val codes: IndexedSeq[EmitCode]) extends SBaseStructCode {
+
+  def code: Code[_] = throw new UnsupportedOperationException
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/types/physical/stypes/concrete/SStackStruct.scala:132]: This should be unnecessary.

**Review by chrisvittal — CHANGES_REQUESTED** [review]

**Comment by tpoterba** [comment]
Bump. This is ready to go and is the base of the SType stack I have open.

**Review by chrisvittal — APPROVED** [review]
LTGM

**Pull request merged**